src/metabase/api/search.clj

```diff
@@ -149,13 +154,22 @@
       (merge-name-and-archived-search search-ctx)
       (add-collection-criteria :collection_id search-ctx)))
 
+(defn- add-child-collection-criteria
+  "Adds a query clause that will include direct dedendents of `parent-coll-id`"
+  [collection-query {parent-coll-id :collection}]
+  (let [child-loc (coll/children-location (if (= :root parent-coll-id)
+                                            coll/root-collection
+                                            (Collection parent-coll-id)))]
+    (h/merge-where collection-query [:= :location child-loc])))
```

Comment:
I've mentioned this a few times but based on the way @mazameli said things should work I think we want "effective children" of a Collection instead of actual children. For example if we have Collections like

```
A > B > C
```

and I can see A and C, but not B, C should "effectively" be a child of A to me (i.e., it would appear to me as such). In other words, I would see:

```
A > C
```

That's how the other endpoints like `GET /api/colleciton/:id` are already set up to work so I would assume it would make sense to have search work the same way. Thoughts @mazameli @salsakran ?